### PR TITLE
DEV-1225, DEV-1228: Recipient Profile Overview, Get Recipient Children endpoints

### PR DESCRIPTION
**Pull request by dpb-bah**
**High level description:**
* Recipient Profile Overview endpoint displays basic data and totals based on the `recipient_id` and `year` provided.
* Get Recipient Children endpoint provides basic data and totals of the children based on the `DUNS` and the `year` provided. 

**Technical details:**
* Utilizes the RecipientProfile, RecipientLookup, DUNS, Legal Entity, and Universal Transaction tables/matviews
* Fixes a minor bug with the DUNS model (business categories were being ignored due to the `,` after the column is listed. 
* Includes various helpers functions to extract bits of information based on the DUNS, Recipient Hash, Recipient ID, etc.

**Link to JIRA Ticket:**
[DEV-1225](https://federal-spending-transparency.atlassian.net/browse/DEV-1225)
[DEV-1228](https://federal-spending-transparency.atlassian.net/browse/DEV-1228)

**The following are ALL required for the PR to be merged:**
- [x] Backend review completed
- [X] Unit & integration tests updated with relevant test cases
- [X] Matview impact assessment completed
- [x] Frontend impact assessment completed - [Recipient Profile Contract](https://github.com/fedspendingtransparency/usaspending-api/blob/feature-recipient-profile-overview/usaspending_api/api_docs/api_documentation/recipient/recipient_profile.md)  (@ebdabbs @dtrinh50 @kojodoesdesign)
- [X] Data validation completed
- [X] API Performance evaluation completed

#### Recipient Overview - Testing with Prod Test DB
```
{{url}}/api/v2/recipient/duns/6c6105fe-bcd6-e92b-321a-03d231a03761-R/?year=latest
1554ms
{{url}}/api/v2/recipient/duns/0036a0cb-0d88-2db3-59e0-0f9af8ffef57-P/?year=all
1823ms
{{url}}/api/v2/recipient/duns/392052ae-92ab-f3f4-d9fa-b57f45b7750b-C/?year=all
1725ms
{{url}}/api/v2/recipient/duns/0036a0cb-0d88-2db3-59e0-0f9af8ffef57-C/?year=all
1628ms
{{url}}/api/v2/recipient/duns/0b2839c0-b343-8703-a845-9d78e2c2ad3b-P/?year=latest
2806ms
```
#### Get Recipient Children - Testing with Prod Test DB
```
{{url}}/api/v2/recipient/children/001006360/?year=latest
1376ms - 2 children
{{url}}/api/v2/recipient/children/004027553/?year=all
6406ms - 29 children
```

### CODE REVIEW

**Comment by dpb-bah** [comment]
Missing `state_province` in List Recipient Children

**Comment by tony-sappe** [comment]
@dpb-bah For the performance tests did you try a large parent recipient, like Lockheed (ff4ee395-14c4-3e60-c94d-fc763c885f98)?

File: usaspending_api/recipient/v2/views/recipients.py [hunk 1]
```diff
@@ -1,151 +1,279 @@
...
-from usaspending_api.recipient.models import DUNS
+from usaspending_api.awards.v2.filters.matview_filters import matview_search_filter
+from usaspending_api.awards.models_matviews import UniversalTransactionView
+from usaspending_api.recipient.models import RecipientProfile, DUNS
+from usaspending_api.references.models import RecipientLookup, RefCountryCode, LegalEntity
 from usaspending_api.recipient.v2.helpers import validate_year, reshape_filters
 
 logger = logging.getLogger(__name__)
 
+# Recipient Levels
```
Comment (nmonga91) [inline usaspending_api/recipient/v2/views/recipients.py]: Very helpful comments!

**Comment by dpb-bah** [comment]
@tony-sappe Just tested with a big one like Booz and that took about 40 seconds.

File: usaspending_api/api_docs/api_documentation/recipient/recipient_profile.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+# Group Profile Page
+
+These endpoints support the individual Recipient Profile pages that display data for a specific DUNS.
+
+## Recipient Overview [/api/v2/recipient/duns/{recipient_id}/{?year}]
+
+This endpoint returns a high-level overview of a specific recipient, given its id.
+
++ Parameters
+    + `recipient_id`: `0036a0cb-0d88-2db3-59e0-0f9af8ffef57-C` (required, string)
```
Comment (nmonga91) [inline usaspending_api/api_docs/api_documentation/recipient/recipient_profile.md]: can we make an explicit note for adding `-C`, `-P`, or `-R` for the recipient level?

Comment (nmonga91) [reply usaspending_api/api_docs/api_documentation/recipient/recipient_profile.md]: can we also add a sample request for each endpoint added here?

Comment (dpb-bah) [reply usaspending_api/api_docs/api_documentation/recipient/recipient_profile.md]: Added both

Comment (tony-sappe) [reply usaspending_api/api_docs/api_documentation/recipient/recipient_profile.md]: While it is good to include documentation on the expected data, the client should never need to create the ID for searches. The API will provide the correct `recipient_id` and the client will pass it back when appropriate.

File: usaspending_api/recipient/models.py [hunk 3]
```diff
@@ -44,7 +44,7 @@ class DUNS(models.Model):
     zip4 = models.TextField(null=True, blank=True)
     country_code = models.TextField(null=True, blank=True)
     congressional_district = models.TextField(null=True, blank=True)
-    business_types_codes = ArrayField(base_field=models.TextField(), default=list, size=None),
```
Comment (tony-sappe) [inline usaspending_api/recipient/models.py]: Yay

**Pull request merged**